src/main/java/com/researchspace/protocolsio/PIOTemperatureComponentType.java

```diff
@@ -1,11 +1,35 @@
 package com.researchspace.protocolsio;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 @Data
+@AllArgsConstructor
+@NoArgsConstructor
 public class PIOTemperatureComponentType {
 
-	private String unit, title;
-	private Integer temperature;
+	private String unit;
+	private String title;
+	private Double temperature;
+
+	@JsonIgnore
+	public String getTemperatureString() {
+		StringBuilder result = new StringBuilder();
+		if(unit != null || !(unit.isEmpty())) {
```

Comment:
this condition uses ||, but others use &&, is this correct ? if unit is null here it will throw an NPE here?

Reply:
This will throw a NPE if unit is null?